tests/common.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+// For some reason clippy & check think that the public methods in this file are not used
```

Comment:
Code in the `tests` folder is treated differently than normal source files in some respects. The [documentation](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory) give more details, and also talks about the case when there's a `common` submodule just like we're trying to do here. I haven't tried it out, but maybe it fixes this problem.

Reply:
it doesn't look like it is fixing the problem :(